client-api/src/main/java/org/corfudb/runtime/kv/core/DbKey.java

```diff
@@ -0,0 +1,58 @@
+package org.corfudb.runtime.kv.core;
+
+import lombok.AllArgsConstructor;
+import org.corfudb.runtime.Api;
+
+import java.util.UUID;
+import java.util.function.Supplier;
+
+/**
+ * Type contract representing of the key of a key-value associative mapping.
+ */
+public interface DbKey<T> extends Comparable<DbKey<T>>, Supplier<T> {
```

Comment:
Protobuf cannot cover many features we need in code, that's why we have these classes.

1. For instance, just using protobuf `Key` class instead of `DbKey` we can't have it as a `Comparable` class.
 
2. Also, there could be many other places in Corfu where we don't need protobuf messages to save a message. Imagine you have `CorfuTable<Integer, String>`, so if we hardcode protobuf in Crud it wouldn't be possible to use Crud for saving data into that table, because we are limited by protobuf messages.

3. Also, there is some code we need to have in `DbKey` to support serialization/deserialization